async/component.go

```diff
@@ -38,36 +41,102 @@ type Component struct {
 	retryWait    time.Duration
 }
 
-// New returns a new async component. The default behavior is to return a error of failure.
-// Use options to change the default behavior.
-func New(name string, p ProcessorFunc, cf ConsumerFactory, oo ...OptionFunc) (*Component, error) {
+// Builder gathers all required properties in order to construct a component
+type Builder struct {
+	errors       []error
+	name         string
+	proc         ProcessorFunc
+	failStrategy FailStrategy
+	cf           ConsumerFactory
+	retries      uint
+	retryWait    time.Duration
+}
 
+// New initializes a new builder for a component with the given name
+// by default the failStrategy will be NackExitStrategy
+func New(name string) *Builder {
+	var errs []error
 	if name == "" {
-		return nil, errors.New("name is required")
+		errs = append(errs, errors.New("name is required"))
 	}
+	return &Builder{
+		name:   name,
+		errors: errs,
+	}
+}
 
-	if p == nil {
-		return nil, errors.New("work processor is required")
+// WithProcessor sets the processing logic for the component
+// it will append an error to the builder if the processor is nil
```

Comment:
all comments should end with a period.
```suggestion
// it will append an error to the builder if the processor is nil.
```